File: Exercises2024/Chapter09_various_ensembles/Fortran/2-MCinNPT/Source/writepdb.f
      Subroutine Writepdb
      Implicit None

      Include 'system.inc'
      
      Integer I,Countmodel
      
      Data Countmodel/ 0/

      Save Countmodel

      Countmodel = Countmodel + 1

      Write(22,'(A,I9)') 'MODEL',Countmodel
      Write(22,'(A6, 3f9.3, 3f7.2, 1x, A)') 'CRYST1',
     &   2.0d0 * Box, 2.0d0 * Box, 2.0d0 * Box,
     &   90.0, 90.0, 90.00, 'P 1         1'

      Do I=1,NumberOfParticles

         Write(22,'(A6,I5,1x,A4,1x,3x,1x,4x,1x,4x,3f8.3,12x,10x,A2)')
     &        'ATOM  ',I,' H  ',
     &        2.0d0 * X(I),2.0d0 * Y(I), 2.0d0 * Z(I),' H'

      Enddo

      Write(22,'(A)') 'ENDMDL'

      Return
      End

File: Exercises2024/Chapter04_basic_mc/C/4-MCLennardJones/Source/writepdb.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include "system.h"


void WritePdb(FILE *FilePtr)
{
  int i;
  static int Countmodel=0;

  Countmodel++;

  fprintf(FilePtr,"%s %9d\n","MODEL", Countmodel);
  fprintf(FilePtr,"CRYST1   %6.3f   %6.3f   %6.3f  90.00  90.00  90.00 P 1         1\n", 2.0*Box, 2.0*Box, 2.0*Box);

  for(i=0;i<NumberOfParticles;i++)
  {
    fprintf(FilePtr,"%s%7d%s%12d    %8.3lf%8.3lf%8.3lf                      %2s\n",
      "ATOM",i,"  H",i,Positions[i].x*2.0,Positions[i].y*2.0,Positions[i].z*2.0," H");
  }
  fprintf(FilePtr,"%s\n","ENDMDL");
}

void Store(void)
{
  int i;
  FILE *FilePtr;

  FilePtr=fopen("restart.dat","w");
  fprintf(FilePtr,"%lf\n",Box);
  fprintf(FilePtr,"%d\n",NumberOfParticles);
  fprintf(FilePtr,"%lf\n",MaximumDisplacement);

  for(i=0;i<NumberOfParticles;i++)
    fprintf(FilePtr,"%lf %lf %lf\n",
      Positions[i].x,Positions[i].y,Positions[i].z);
  fclose(FilePtr);
}

